procedures.sql

```diff
@@ -0,0 +1,1136 @@
+    @webinar_id INT,
+    @payment_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentWebinar
+    SET payment_date = @payment_date
+    WHERE student_id = @student_id AND webinar_id = @webinar_id;
+END;
+GO
+-- EXEC PayForWebinar 17, 23
+
+
+
+CREATE PROCEDURE AddStudentToSemester
+    @student_id INT,
+    @semester_id INT,
+    @payment_date DATETIME = NULL
+AS
+BEGIN
+    INSERT INTO StudentSemester (student_id, semester_id, payment_date)
+    VALUES (@student_id, @semester_id, @payment_date);
+END;
+GO
+-- EXEC AddStudentToSemester 17, 23
+
+CREATE PROCEDURE RemoveStudentFromSemester
+    @student_id INT,
+    @semester_id INT
+AS
+BEGIN
+    DELETE FROM StudentSemester
+    WHERE student_id = @student_id AND semester_id = @semester_id;
+END;
+GO
+-- EXEC RemoveStudentFromSemester 17, 23
+
+CREATE PROCEDURE PayForSemester
+    @student_id INT,
+    @semester_id INT,
+    @payment_date DATETIME = CURRENT_TIMESTAMP
+AS
+BEGIN
+    UPDATE StudentSemester
+    SET payment_date = @payment_date
+    WHERE student_id = @student_id AND semester_id = @semester_id;
+END;
+GO
+-- EXEC PayForSemester 17, 23, @payment_date = '2021-01-01 12:00:00'
+
+
+
+CREATE PROCEDURE AddStudentToInternship
+    @student_id INT,
+    @internship_id INT
+AS
+BEGIN
+    INSERT INTO InternshipStudent (student_id, internship_id)
+    VALUES (@student_id, @internship_id);
+END;
+GO
+-- EXEC AddStudentToInternship 17, 23
+
+CREATE PROCEDURE RemoveStudentFromInternship
+    @student_id INT,
+    @internship_id INT
+AS
+BEGIN
+    DELETE FROM InternshipStudent
+    WHERE student_id = @student_id AND internship_id = @internship_id;
+END;
+GO
+-- EXEC RemoveStudentFromInternship 17, 23
+
+CREATE PROCEDURE LogInternshipAttendance
+    @student_id INT,
+    @internship_id INT
+AS
+BEGIN
+    UPDATE InternshipStudent
+    SET attended_days = attended_days + 1
+    WHERE student_id = @student_id AND internship_id = @internship_id;
+END;
+GO
+-- EXEC LogInternshipAttendance 17, 23
+
+CREATE PROCEDURE SetInternshipExamResult
+    @student_id INT,
+    @internship_id INT,
+    @exam_result INT
+AS
+BEGIN
+    UPDATE InternshipStudent
+    SET exam_result = @exam_result
+    WHERE student_id = @student_id AND internship_id = @internship_id;
+END;
+GO
+-- EXEC SetInternshipExamResult 17, 23, 60
+
+
+
+CREATE PROCEDURE LogMeetingAttendance
+    @student_id INT,
+    @meeting_id INT
+AS
+BEGIN
+    INSERT INTO StudentMeetingAttendance (student_id, meeting_id)
+    VALUES (@student_id, @meeting_id);
+END;
+GO
+-- EXEC LogMeetingAttendance 17, 23
+
+
+
+CREATE PROCEDURE AddParameter
+    @name NVARCHAR(50),
+    @value NVARCHAR(50),
+    @date DATE = NULL
+AS
+BEGIN
+    IF @date IS NULL
+        SET @date = CURRENT_TIMESTAMP;
+
+    INSERT INTO Parameter (name, value, date)
+    VALUES (@name, @value, @date);
+END;
+GO
+-- EXEC AddParameter 'TEST PARAMETER', 2, '2019-01-01';
+-- EXEC AddParameter 'TEST PARAMETER', 'value';
+
+
+
+CREATE PROCEDURE CreateBasket
```

Comment:
what is the difference between `adding` and `creating`? It must be if you use both of these forms, yup?

Reply:
Definitely, you do not create a student, you add they to the database, but you definitely create a new basket

Reply:
🆗